pkg/operator/controller/route-deprecated/controller.go

```diff
@@ -0,0 +1,227 @@
+	routeConfigAck415Msg   = "A route in this cluster contains a configuration that isn't supported in 4.16. Please review each route's deprecated status condition for more information."
+	adminGateConfigMapName = "admin-gates"
+)
+
+var (
+	log = logf.Logger.WithName(controllerName)
+)
+
+// New creates the route deprecated controller. This is the controller
+// that handles interpreting the route deprecated status into an admin gate
+// which will block upgrades until an admin acks it or resolves the deprecation.
+func New(mgr manager.Manager, config Config) (controller.Controller, error) {
+	operatorCache := mgr.GetCache()
+	reconciler := &reconciler{
+		cache:     config.RouteCache,
+		namespace: config.Namespace,
+		client:    mgr.GetClient(),
+	}
+
+	c, err := controller.New(controllerName, mgr, controller.Options{
+		Reconciler: reconciler,
+		RateLimiter: workqueue.NewMaxOfRateLimiter(
+			// Rate-limit to 1 update every 5 seconds per
+			// reconcile to avoid burning CPU if object
+			// updates are frequent.
+			workqueue.NewItemExponentialFailureRateLimiter(5*time.Second, 30*time.Second),
+			// 10 qps, 100 bucket size, same as DefaultControllerRateLimiter().
+			&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(10), 100)},
+		),
+	})
+
+	// Index routes over the deprecated condition so that we can
+	// efficiently find deprecated routes.
+	if err := reconciler.cache.IndexField(context.Background(), &routev1.Route{}, "deprecatedRoute", client.IndexerFunc(func(o client.Object) []string {
+		if cond := getRouteDeprecated(o.(*routev1.Route)); cond != nil {
+			return []string{string(operatorv1.ConditionTrue)}
+		}
+		return []string{}
+	})); err != nil {
+		return nil, fmt.Errorf("failed to create index for routes: %v", err)
+	}
+
+	// Only reconcile when the route's deprecated status is added, changed, or removed.
+	updatedDeprecatedConditionFilter := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldRoute, ok := e.ObjectOld.(*routev1.Route)
+			if !ok {
+				return false
+			}
+			newRoute, ok := e.ObjectNew.(*routev1.Route)
+			if !ok {
+				return false
+			}
+			return cmp.Equal(getRouteDeprecated(newRoute), getRouteDeprecated(oldRoute))
+		},
+	}
+
+	// Only reconcile when the admin gate config map changed or created.
+	adminGateFilter := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			configmap, ok := e.ObjectNew.(*corev1.ConfigMap)
+			if !ok {
+				return false
+			}
+			return configmap.Name == adminGateConfigMapName && configmap.Namespace == operatorcontroller.GlobalMachineSpecifiedConfigNamespace
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			configmap, ok := e.Object.(*corev1.ConfigMap)
+			if !ok {
+				return false
+			}
+			return configmap.Name == adminGateConfigMapName && configmap.Namespace == operatorcontroller.GlobalMachineSpecifiedConfigNamespace
+		},
+	}
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Please move the blank line.  
```suggestion
	if err != nil {
		return nil, err
	}

```

Reply:
Oh this was completely in the wrong area. Moved.